_posts/2025-02-03-Reduce-Cost-with-Disk-based-Vector-Search.md

```diff
@@ -0,0 +1,251 @@
+| *notes*    | See https://huggingface.co/datasets/Cohere/msmarco-v2.1-embed-english-v3 |
+
+We chose four different cluster configurations to test against and used [opensearch-cluster-cdk](https://github.com/opensearch-project/opensearch-cluster-cdk) and OpenSearch 2.18 to provision the clusters. In general, we chose the cluster and index configuration to follow production recommendations. For instance, we configured replica shards and dedicated cluster manager nodes. In addition, we targeted the shard count to have between 2 and 3 vCPUs per shard.
+
+| Name        | Data node count | Data node type | Data node disk size | Data node disk type | JVM Size | Primary shard count | replica shards | compression level |
+|-------------|-----------------|----------------|---------------------|---------------------|----------|---------------------|----------------|-------------------|
+| in_memory   | 8               | r6g.8xlarge    | 300                 | EBS                 | 32       | 40                  | 1              | 1x                |
+| on_disk_8x  | 10              | r6gd.2xlarge   | 474                 | Instance            | 32       | 15                  | 1              | 8x                |
+| on_disk_16x | 6               | r6gd.2xlarge   | 474                 | Instance            | 32  9    | 1                   | 16x            |
+| on_disk_32x | 4               | r6gd.2xlarge   | 474                 | Instance            | 32  6    | 1                   | 32x            |
+
+From then above table, you can see that the higher level of compression clusters use substantially less resources than the non-compressed.
+
+In order to run the tests, we also made a few other optimizations to tune performance:
+
+1. Retrieved ids from doc values in order to reduce fetch time
+2. Disabled source storage 
+
+The test procedure was fairly simple. All tests were executed with OSB. See the [vectorsearch workloads](https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/main/vectorsearch) for more details.  We did the following:
+
+1. Ingest the data set
+2. Force merged to 5 segments per shard
+3. Warmed the index up
+4. Ran the Single-client latency test
+5. Ran the Multi-client throughput tests
+6. Repeated 4 and 5 with disk-based re-score setting disabled to show what the results would be without re-scoring, effectively searching over a compressed index without re-scoring.
+
+#### Results
+
+From these experiments, we obtained the following results:
+
+| Metric/Configuration              | in-memory | on_disk_8x | in_memory_8x | on_disk_16x | in_memory_16x | on_disk_32x | in_memory_32x |
+|-----------------------------------|-----------|------------|--------------|-------------|---------------|-------------|---------------|
+| recall@100 (ratio)                | 0.95      | 0.98       | 0.98         | 0.97        | 0.96          | 0.94        | 0.95          |
```

Comment:
This shows that 32x compression just works and we should not add on_disk here

Reply:
Added both just for sake of transparency. For some data sets, the re-scoring does not significantly help.